tests/components/modbus/conftest.py

```diff
@@ -73,14 +91,11 @@ async def run_base_test(
     else:  # CALL_TYPE_REGISTER_HOLDING
         use_mock_hub.read_holding_registers.return_value = read_result
 
-    # Initialize sensor
-    now = dt_util.utcnow()
-    with mock.patch("homeassistant.helpers.event.dt_util.utcnow", return_value=now):
-        assert await async_setup_component(hass, entity_domain, config)
-        await hass.async_block_till_done()
-
     # Trigger update call with time_changed event
-    now += timedelta(seconds=scan_interval + 1)
     with mock.patch("homeassistant.helpers.event.dt_util.utcnow", return_value=now):
         async_fire_time_changed(hass, now)
         await hass.async_block_till_done()
+
+    # Check state
+    state = hass.states.get(entity_id).state
+    assert state == expected
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6269273)